### ISSUE TITLE
Weights noise in Keras

### ISSUE THREAD

**Issue by Jihadik** [fchollet/keras#2803]
Is there any way of adding noise to the weights of your network after each batch iteration in Keras?
If there is no what is the best way of doing that?

**Comment by the-moliver** [comment]
That isn't currently implemented but it would be easy to write your own layer to do it. Look at the code for ```Dense``` and the layers in the ```Noise``` module. Dropout and GaussianDropout put multiplicative noise on the activations which are determined by the weights, so I'm not sure what the advantage of adding noise directly to the weights would be. Is there a particular reason you want to do that?

**Comment by kkawabat** [comment]
@the-moliver There has been some research in injecting weighting noise during training of NN to prevent overfitting of data. (source: http://www.ncbi.nlm.nih.gov/pubmed/19928111)

**Comment by the-moliver** [comment]
@kkawabat I'm aware of that, but most of that work is before the advent of dropout. I was only saying its unclear if there's an advantage of weight noise over activation noise (i.e. dropout/gaussiandropout)

**Comment by Jihadik** [comment]
> Is there a particular reason you want to do that?
In fact it was said by one of the postdocs in MILA Lab that it works in Speech Recognition. I tried and didn't find any profit but anyway.

**Comment by shi27feng** [comment]
Hi,
I'm also currently trying to add some noise on weights, have you found any solution? @Jihadik
Thanks

**Comment by Jihadik** [comment]
I was able to do so, but don't remember, was it in Keras or Tensorflow. It was a long time ago :)
